infrastructure/zk/src/docker.ts

```diff
@@ -138,6 +143,10 @@ export async function build(image: string, cmd: Command) {
     await dockerCommand('build', image, cmd.customTag);
 }
 
+export async function customBuildForHyperchain(image: string, imageName: string) {
+    await dockerCommand('build', image, '', false, imageName);
+}
```

Comment:
Are we OK with the fact that we can build the hyperchain image via `zk`, but cannot push?
Right now it's hard for me to comprehend how will we introduce a generic solution on pushing images, since we have a significant number of performance optimizations implemented to make prover images work fast.

What's the plan here?

Reply:
I think it is totally ok. Generic pushing would be too complex for now. If someone knows how to build, they most likely can figure out pushing the tagged image themselves

Reply:
But then wouldn't it make more sense for them to also build the images themselves? E.g. they may want to have their own tagging logic.
My main concern is that this functionality doesn't seem to be fitting; e.g. it does *half* the job, and if you still have to do the other half youself, then why bother? Overall, all it does is passes some flags to the `docker build` command; might have been simply covered in docs. This way it will probably be less confusing, e.g. "you build -> you push" instead of "zk does some magic -> you have to reverse-engineer this magic -> you push".

Reply:
But the build part is more than just firing the command, it does have some specific logic for some components. But you do make a good point on making it more customizable. Changed a little bit, let me know what you think

Reply:
Developer is only forced to use it If they are customizing their hyperchain. So they need to build the image.

Reply:
No objections, just pointing out that we may improve things on this front to be friendlier to the developers.